src/parsers/cdm_parser.jl

```diff
@@ -148,93 +252,108 @@ function gen_csv_parser(gen_raw::DataFrames.DataFrame, Buses::Dict{Int64,Any})
+        hr = [n for n in names(gen_raw) if !isa(match(Regex("HR_.*_$i"),String(n)),Nothing)][1]
+        mw = Symbol("Output_pct_$i")
+        push!(cost_colnames, (mw,hr))
+    end
+
     for gen in 1:nrow(gen_raw)
+        pmax = tryparse(Float64,"""$(gen_raw[gen,Symbol("PMax MW")])""")
+
         if gen_raw[gen,:Fuel] in ["Oil","Coal","NG","Nuclear"]
-            var_cost = [float(gen_raw[gen,i]) for i in 31:40]
-            fuel_cost = gen_raw[gen,30]./1000
-            bus_id =[Buses[i] for i in keys(Buses) if Buses[i]["number"] == gen_raw[gen,2]]
-            Generators_dict["Thermal"][gen_raw[gen,1]] = Dict{String,Any}("name" => gen_raw[gen,1],
+
+            fuel_cost = gen_raw[gen,Symbol("Fuel Price \$/MMBTU")]./1000
+
+            var_cost = [(tryparse(Float64,"$(gen_raw[gen,cn[1]])"), tryparse(Float64,"$(gen_raw[gen,cn[2]])")) for cn in cost_colnames]
+            var_cost = [(c[1],c[1]*c[2]*fuel_cost).*pmax for c in var_cost if !in(nothing,c)]
+            var_cost[2:end] = [(var_cost[i][1],var_cost[i-1][2]+var_cost[i][2]) for i in 2:length(var_cost)]
+
+            bus_id =[Buses[i] for i in keys(Buses) if Buses[i]["number"] == gen_raw[gen,Symbol("Bus ID")]]
+            Generators_dict["Thermal"][gen_raw[gen,Symbol("GEN UID")]] = Dict{String,Any}("name" => gen_raw[gen,Symbol("GEN UID")],
                                             "available" => true,
                                             "bus" => make_bus(bus_id[1]),
-                                            "tech" => Dict{String,Any}("activepower" => 0,
-                                                                        "activepowerlimits" => (min=float(gen_raw[gen,12]),max=float(gen_raw[gen,11])),
-                                                                        "reactivepower" => 0,
-                                                                        "reactivepowerlimits" => (min=float(gen_raw[gen,14]),max=float(gen_raw[gen,13])),
-                                                                        "ramplimits" => (up=gen_raw[gen,17],down=gen_raw[gen,17]),
-                                                                        "timelimits" => (up=gen_raw[gen,16],down=gen_raw[gen,15])),
-                                            "econ" => Dict{String,Any}("capacity" => gen_raw[gen,11],
+                                            "tech" => Dict{String,Any}("activepower" => gen_raw[gen,Symbol("MW Inj")],
+                                                                        "activepowerlimits" => (min=tryparse(Float64,"""$(gen_raw[gen,Symbol("PMin MW")])"""),max=pmax),
+                                                                        "reactivepower" => tryparse(Float64,"""$(gen_raw[gen,Symbol("MVAR Inj")])"""),
+                                                                        "reactivepowerlimits" => (min=tryparse(Float64,"""$(gen_raw[gen,Symbol("QMin MVAR")])"""),max=tryparse(Float64,"""$(gen_raw[gen,Symbol("QMax MVAR")])""")),
```

Comment:
These type conversions to Symbol can be expensive computationally. Is there a reason to use symbols and not strings in the resulting dict?

Reply:
In this particular case, the symbol conversion is to access the column headers in a dataframe which are symbols. The previous implementation used numeric column indices that were hard-coded. Perhaps the better approach would be to create a column index map so that numeric indices are used to access columns, but the parser doesn't depend on a certain order of columns.... this could also be made a bit more flexible to accommodate different column names.

Reply:
only RTS... seems fine so far. I'll implement my suggested changes. There are a few areas that I'm worried about performance, but those can be addressed as we scale up.